source/sigv4.c

```diff
@@ -1897,36 +1918,40 @@ static SigV4Status_t generateCredentialScope( const SigV4Parameters_t * pSigV4Pa
                 pBufLoc += encodedLen;
                 remainingLen -= encodedLen;
 
-                /* An empty value corresponds to an empty string. */
-                if( pCanonicalRequest->pQueryLoc[ i ].value.dataLen > 0U )
+                /* Encode parameter value if non-empty. Query parameters can have empty values. */
+                if( pCanonicalRequest->pQueryLoc[ paramsIndex ].value.dataLen > 0U )
                 {
-                    assert( pCanonicalRequest->pQueryLoc[ i ].value.pData != NULL );
+                    assert( pCanonicalRequest->pQueryLoc[ paramsIndex ].value.pData != NULL );
                     returnStatus = writeValueInCanonicalizedQueryString( pBufLoc,
                                                                          remainingLen,
-                                                                         pCanonicalRequest->pQueryLoc[ i ].value.pData,
-                                                                         pCanonicalRequest->pQueryLoc[ i ].value.dataLen,
+                                                                         pCanonicalRequest->pQueryLoc[ paramsIndex ].value.pData,
+                                                                         pCanonicalRequest->pQueryLoc[ paramsIndex ].value.dataLen,
                                                                          &encodedLen );
                     pBufLoc += encodedLen;
                     remainingLen -= encodedLen;
                 }
             }
 
-            if( ( remainingLen < 1U ) && ( numberOfParameters != ( i + 1U ) ) )
+            /* Check if there is a next parameter. */
+            thereExistsNextParameter = ( ( paramsIndex + 1U ) != numberOfParameters );
+
+            /* Before adding the '&' for the next query parameter, check that this is not
+             * the last parameter. */
+            if( thereExistsNextParameter && ( remainingLen > 0U ) )
```

Comment:
I would suggest this as it avoids the empty `else`
```c
if( thereExistsNextParameter )
{
    if( remainingLen > 0U )
    {
        *pBufLoc = '&';
        ++pBufLoc;
        remainingLen--;
    }
    else
    {
        returnStatus = SigV4InsufficientMemory;
        LOG_INSUFFICIENT_MEMORY_ERROR( "write the canonical query", 1U );
        break;
    }
}
```

Reply:
Adding nesting increases complexity